interview-questions.md

```diff
@@ -8,36 +8,58 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: Rails new, creates a new application. The db migrate allow edits like addition to add columns to a current database. The db generate, is used to create a new database.
 
-Researched answer:
+Researched answer: Rails new creates a new rails application, installs the boilerplate code and framework in the directory that's created along with the command. Files are automatically installed for a Rails project called 'gems'. The db migrate is a feature of Active Record that allows you to make different types of changes to an exsisting database and effects the database schema over a period of time, changes include updating key-value property, add and delete columns. The db generates commands follows the MVC model, view and controller. This is the link between the UI and the database.
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET" location: /students
+action: "GET" location: /students Points to the index in the controller method
 
-action: "POST" location: /students
+action: "POST" location: /students Points to forward/create which adds additional content
 
-action: "GET" location: /students/new
+action: "GET" location: /students/new Defines a new function, there would be a form for the student.
 
-action: "GET" location: /students/2
+action: "GET" location: /students/2 Directed at the show method of the location - student/2
 
-action: "GET" location: /students/2/edit
+action: "GET" location: /students/2/edit Edit method contents located in - student/2
 
-action: "PATCH" location: /students/2
+action: "PATCH" location: /students/2 Updates to the student/2 location
 
-action: "DELETE" location: /students/2
+action: "DELETE" location: /students/2 Destroy contents in student/2
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+
+1. Create a new project/ rails application break down tasks into small tasks that you can test inorder to make sure the application is working fine.
+
+2. Create a database and make configurations to the specs of the title of the title of the appliction
+
+3. Create models for the required database.
+
+4. Create drawouts for the actual design of the application
+
+5. Create controllers with simple actions.
+
+6. Create the logic for the application.
+
+7. Create routes/ UI for the front-end (actaul page) of the application.
+
+8. Deploy and navigate website to test functionality.
+
+9. Get feedback on the actual site on whats working versus what needs work and additions if any is required from the actual consumers.
+
+10. Fix any bugs/ add additions according to the feedback.
```

Comment:
Good job at breaking down the higher level item necessary to make the app.  For user stories, you want to breakdown into specific tasks.  Like the Trello boards we are using right now.  :)